### DISCUSSION TITLE
RAG on Cosmos DB MongoDB: Session Resources

### DISCUSSION THREAD

**Discussion by multispark** [microsoft/RAG_Hack#48] [Announcements]
This is the resources and discussion thread for this live stream: https://aka.ms/raghack/cosmosdb
It will be updated with links after the session.

Recording: https://www.youtube.com/live/DnvmVRFffMY
Slides: https://github.com/microsoft/RAG_Hack/blob/main/slides/RAGHackCosmosDBMongo.pptx

Please ask any follow-up questions here, and the speakers will see them.

**Comment by khelanmodi** [comment]
Thanks for joining the session! Here are the links to all the useful resources we discussed:
- Azure Cosmos DB for MongoDB vCore Free tier: [Aka.ms/tryvcore​](https://aka.ms/tryvcore)
- Demo code: [Aka.ms/vcorelangchain​](https://aka.ms/vcorelangchain)
- Additional samples, docs and more: [Aka.ms/CosmosAISamples](https://github.com/AzureCosmosDB/AISamples)

Feel free to ask any questions on this thread!

**Reply by zzulueta** [reply]
The previous RAG code/demos for Azure Cosmos DB MongoDb vCore used Semantic Kernel in building the index. Why did we switch to Langchain?

**Comment by john0isaac** [comment]
Thanks for joining us! in case the shortened link doesn't work here is the demo code URL:
https://github.com/Azure-Samples/Cosmic-Food-RAG-app

**Reply by zzulueta** [reply]
You mentioned in this code below that you can only run this once as it creates the embeddings. 
- If you run it again does it recreate new embeddings and hence makes double entries?
- How do you handle the scenario that we have new entries to the database and we just want to compute for the embeddings for those new entries?

# Only Run this the first time you open the notebook
# Create embeddings from the data, save to the database and return a connection to MongoDB vCore
vector_store: AzureCosmosDBVectorSearch = AzureCosmosDBVectorSearch.from_documents(
    json_data,
    azure_openai_embeddings,
    collection=collection,
    index_name=index_name,
)

**Reply by john0isaac** [reply]
@zzulueta Thank you for your question.

- Yes, if you run it using the same sample of data it will recreate new embeddings from the same data and populate the database with duplicate entries.
- Don't run it using the same sample of data. if you have new entries run it using the new entries not the entire data sample.

If you have anymore questions don't hesitate to ask.
